sechub-scan/src/test/java/com/daimler/sechub/domain/scan/HTMLScanResultReportModelBuilderTest.java

```diff
@@ -23,145 +25,181 @@
+        assertNotNull(map.get("includedCSSRef"));
+        String path = (String) map.get("includedCSSRef");
+        File foundFile = new File(path);
+
+        assertEquals(expectedFile.getCanonicalPath(), foundFile.getCanonicalPath());
+
+        assertEquals("trafficlight1", map.get("trafficlight"));
+        assertEquals(uuid.toString(), map.get("jobuuid"));
+        assertEquals(HIDE_LIGHT, map.get("styleRed"));
+        assertEquals(HIDE_LIGHT, map.get("styleYellow"));
+        assertEquals(HIDE_LIGHT, map.get("styleGreen"));
+    }
+
+    @Test
+    public void trafficlight_red_set_display_block__others_are_none() {
+        when(scanResult.getTrafficLight()).thenReturn(TrafficLight.RED.name());
+
+        Map<String, Object> map = builderToTest.build(scanResult);
+        assertEquals(SHOW_LIGHT, map.get("styleRed"));
+        assertEquals(HIDE_LIGHT, map.get("styleYellow"));
+        assertEquals(HIDE_LIGHT, map.get("styleGreen"));
+    }
+
+    @Test
+    public void trafficlight_yellow_set_display_block__others_are_none() {
+        when(scanResult.getTrafficLight()).thenReturn(TrafficLight.YELLOW.name());
+
+        Map<String, Object> map = builderToTest.build(scanResult);
+        assertEquals(HIDE_LIGHT, map.get("styleRed"));
+        assertEquals(SHOW_LIGHT, map.get("styleYellow"));
+        assertEquals(HIDE_LIGHT, map.get("styleGreen"));
+    }
+
+    @Test
+    public void trafficlight_green_set_display_block__others_are_none() {
+        when(scanResult.getTrafficLight()).thenReturn(TrafficLight.GREEN.name());
+
+        Map<String, Object> map = builderToTest.build(scanResult);
+        assertEquals(HIDE_LIGHT, map.get("styleRed"));
+        assertEquals(HIDE_LIGHT, map.get("styleYellow"));
+        assertEquals(SHOW_LIGHT, map.get("styleGreen"));
+    }
+
+    @Test
+    public void test_code_scan_entries() {
+
+        SecHubFinding finding = mock(SecHubFinding.class);
+        SecHubCodeCallStack code1 = mock(SecHubCodeCallStack.class);
+        SecHubCodeCallStack subCode = mock(SecHubCodeCallStack.class);
+
+        when(scanResult.getTrafficLight()).thenReturn(TrafficLight.RED.name());
+        when(scanResult.getResult()).thenReturn(result);
+        when(result.getFindings()).thenReturn(Arrays.asList(finding));
+        when(finding.getCode()).thenReturn(code1);
+        when(code1.getCalls()).thenReturn(subCode);
+
+        Map<String, Object> buildResult = builderToTest.build(scanResult);
+
+        assertNotNull(buildResult.get("codeScanEntries"));
+
+        assertTrue(buildResult.get("codeScanEntries") instanceof Map<?, ?>);
+
+        Map<Integer, List<HTMLScanResultCodeScanEntry>> codeScanEntries = (Map<Integer, List<HTMLScanResultCodeScanEntry>>) buildResult.get("codeScanEntries");
+        assertNotNull(codeScanEntries.get(0));
+
+        List<HTMLScanResultCodeScanEntry> scanEntriesList = codeScanEntries.get(0);
+        assertTrue(scanEntriesList.size() == 2);
+    }
+
+    @Test
```

Comment:
Could you please add

/* prepare */

/* execute */

/* test */

Reply:
test_code_scan_support -> would the method be more a "when_scan_result_has_red_trafficlight_built_map_has_codeScanSupport_object_not_null" ?

Or shall this happen always (even when no traffic light set) or only when red?
Sorry but I don't get it.

Reply:
done